### PR TITLE
Optimise test framework (bunq/sdk_java#78)

### PR DESCRIPTION
**Pull request by OGKevin**
References #78

### CODE REVIEW

File: src/main/java/com/bunq/sdk/http/ApiClient.java [hunk 1]
```diff
@@ -85,15 +85,15 @@
...
+  public static final String CACHE_CONTROL_NONE = "no-cache";
 
   /**
    * Prefix for bunq's own headers.
    */
   private static final String USER_AGENT_BUNQ = "bunq-sdk-java/0.13.0";
-  private static final String LANGUAGE_EN_US = "en_US";
-  private static final String REGION_NL_NL = "nl_NL";
-  private static final String GEOLOCATION_ZERO = "0 0 0 0 000";
+  public static final String LANGUAGE_EN_US = "en_US";
```
Comment (patrickdw1991) [inline src/main/java/com/bunq/sdk/http/ApiClient.java]: public under private?

File: src/test/java/com/bunq/sdk/BunqSdkTestBase.java [hunk 2]
```diff
@@ -4,51 +4,190 @@
...
    */
+  private static final String ERROR_COULD_NOT_GENERATE_NEW_API_KEY = "Encountered error while retrieving new sandbox ApiKey.\nError message %s";
+
   private static final String DEVICE_DESCRIPTION = "Java test device";
+  private static final int HTTP_STATUS_OK = 200;
+  private static final String FIELD_RESPONSE = "Response";
+  private static final String FIELD_API_KEY = "ApiKey";
+  private static final int INDEX_FIRST = 0;
+
+  protected static final String TEST_CONFIG_PATH = "bunq-test.conf";
```
Comment (patrickdw1991) [inline src/test/java/com/bunq/sdk/BunqSdkTestBase.java]: protected under private?

File: src/test/java/com/bunq/sdk/BunqSdkTestBase.java [hunk 3]
```diff
@@ -4,51 +4,190 @@
...
 
   @BeforeClass
-  public static void setUp() {
+  public static void setUpBeforeClass() {
     BunqContext.loadApiContext(getApiContext());
+    setSecondMonetaryAccountBank();
+    requestSpendingMoney();
+
+    try {
+      Thread.sleep(500);
```
Comment (patrickdw1991) [inline src/test/java/com/bunq/sdk/BunqSdkTestBase.java]: Why this sleep?

File: src/test/java/com/bunq/sdk/BunqSdkTestBase.java [hunk 4]
```diff
@@ -4,51 +4,190 @@
...
+  }
+
+  private static void setSecondMonetaryAccountBank() {
+    BunqResponse<Integer> response = MonetaryAccountBank.create(CURRENCY_EUR, ACCOUNT_DESCRIPTION);
 
-    apiContext.save(apiConfigPath);
+    secondMonetaryAccountBank = MonetaryAccountBank.get(response.getValue()).getValue();
+  }
+
+  private static void requestSpendingMoney() {
```
Comment (patrickdw1991) [inline src/test/java/com/bunq/sdk/BunqSdkTestBase.java]: People using the SDK might not really understand what's actually happening here, maybe just add a comment.

File: src/test/java/com/bunq/sdk/BunqSdkTestBase.java [hunk 5]
```diff
@@ -4,51 +4,190 @@
...
+          new Pointer(POINTER_TYPE_EMAIL, SUGAR_DADDY_EMAIL),
+          SUGAR_DADDY_REQUESTS_DESCRIPTION,
+          false,
+          secondMonetaryAccountBank.getId()
+      );
+    }
   }
 
-} 
+  protected static Pointer getPointerBravo() {
```
Comment (patrickdw1991) [inline src/test/java/com/bunq/sdk/BunqSdkTestBase.java]: And wasn't the convention to always add (empty) docblocks?

File: src/test/java/com/bunq/sdk/BunqSdkTestBase.java (hunk 4 above)
Comment (OGKevin) [reply src/test/java/com/bunq/sdk/BunqSdkTestBase.java]: but, this is internal lib ? User shouldn't even know whats is happening here ? 🤔

File: src/test/java/com/bunq/sdk/BunqSdkTestBase.java (hunk 5 above)
Comment (OGKevin) [reply src/test/java/com/bunq/sdk/BunqSdkTestBase.java]: i thought we agreed to there is no need for it anymore 🤔 as it didn't bring any value as its a strict typed lang.

File: src/test/java/com/bunq/sdk/BunqSdkTestBase.java (hunk 4 above)
Comment (patrickdw1991) [reply src/test/java/com/bunq/sdk/BunqSdkTestBase.java]: It's open source. People might want to look into why a request was randomly created and paid.

File: src/test/java/com/bunq/sdk/BunqSdkTestBase.java (hunk 3 above)
Comment (OGKevin) [reply src/test/java/com/bunq/sdk/BunqSdkTestBase.java]: To ensure the spending money request has been accepted before refreshing. The refresh part will be added with #79

File: src/test/java/com/bunq/sdk/BunqSdkTestBase.java (hunk 4 above)
Comment (OGKevin) [reply src/test/java/com/bunq/sdk/BunqSdkTestBase.java]: in the tests ? :P

File: src/test/java/com/bunq/sdk/BunqSdkTestBase.java (hunk 5 above)
Comment (patrickdw1991) [reply src/test/java/com/bunq/sdk/BunqSdkTestBase.java]: We agreed on being consistent I think. I have posted multiple times that they where not there and you added it multiple times as that was the way you did it before. Switching styles all the time is messy, do it or don't but make sure to do it consistent.

Comment (OGKevin) [reply src/test/java/com/bunq/sdk/BunqSdkTestBase.java]: Alrighty then, as its a strict lang, no need for empty doc blocks. in the non strict ones will add doc blocks to define parameters etc.

File: src/test/java/com/bunq/sdk/BunqSdkTestBase.java (hunk 4 above)
Comment (patrickdw1991) [reply src/test/java/com/bunq/sdk/BunqSdkTestBase.java]: In the test. We actually mention how to run the tests in the README.md. 

In this case the method name is clear but the process is quite weird (the sugar daddy stuff), I'm also not a big fan of comments but somethimes stuff happens outside of the code and it's not clear that's where you want to explain a bit.

This is open source code so people will actually look at it to understand, it's not something internal where we just point to a wiki. Or you can just create a wiki for the github page :P

File: src/test/java/com/bunq/sdk/BunqSdkTestBase.java (hunk 5 above)
Comment (patrickdw1991) [reply src/test/java/com/bunq/sdk/BunqSdkTestBase.java]: Ok so you're gonna change this in the generator?

File: src/test/java/com/bunq/sdk/BunqSdkTestBase.java (hunk 4 above)
Comment (OGKevin) [reply src/test/java/com/bunq/sdk/BunqSdkTestBase.java]: Will write a comment in then 👍

File: src/test/java/com/bunq/sdk/BunqSdkTestBase.java (hunk 5 above)
Comment (OGKevin) [reply src/test/java/com/bunq/sdk/BunqSdkTestBase.java]: yea, but in a future PR then. That might cost some time. Will create followup issue.

Comment (OGKevin) [reply src/test/java/com/bunq/sdk/BunqSdkTestBase.java]: https://github.com/bunq/sdk_java/issues/88

File: src/main/java/com/bunq/sdk/http/ApiClient.java (hunk 1 above)
Comment (OGKevin) [reply src/main/java/com/bunq/sdk/http/ApiClient.java]: hmm., i normally group constants in groups, not on visibility. Error constants on top, ten the first used group in the class. So there could indeed be a mixture of visibility in a specific group.

**Comment by OGKevin** [comment]
@andrederoos

**Pull request merged**